frame/uniques/src/lib.rs

```diff
@@ -0,0 +1,1182 @@
+		/// - `issuer`: The new Issuer of this asset.
+		/// - `admin`: The new Admin of this asset.
+		/// - `freezer`: The new Freezer of this asset.
+		/// - `free_holding`: Whether a deposit is taken for holding an instance of this asset
+		///   class.
+		/// - `is_frozen`: Whether this asset class is frozen except for permissioned/admin
+		/// instructions.
+		///
+		/// Emits `AssetStatusChanged` with the identity of the asset.
+		///
+		/// Weight: `O(1)`
+		#[pallet::weight(T::WeightInfo::force_asset_status())]
+		pub(super) fn force_asset_status(
+			origin: OriginFor<T>,
+			#[pallet::compact] class: T::ClassId,
+			owner: <T::Lookup as StaticLookup>::Source,
+			issuer: <T::Lookup as StaticLookup>::Source,
+			admin: <T::Lookup as StaticLookup>::Source,
+			freezer: <T::Lookup as StaticLookup>::Source,
+			free_holding: bool,
+			is_frozen: bool,
+		) -> DispatchResult {
+			T::ForceOrigin::ensure_origin(origin)?;
+
+			Class::<T, I>::try_mutate(class, |maybe_asset| {
+				let mut asset = maybe_asset.take().ok_or(Error::<T, I>::Unknown)?;
+				asset.owner = T::Lookup::lookup(owner)?;
+				asset.issuer = T::Lookup::lookup(issuer)?;
+				asset.admin = T::Lookup::lookup(admin)?;
+				asset.freezer = T::Lookup::lookup(freezer)?;
+				asset.free_holding = free_holding;
+				asset.is_frozen = is_frozen;
+				*maybe_asset = Some(asset);
+
+				Self::deposit_event(Event::AssetStatusChanged(class));
+				Ok(())
+			})
+		}
+
+		/// Set the metadata for an asset instance.
+		///
+		/// Origin must be either `ForceOrigin` or Signed and the sender should be the Owner of the
+		/// asset `class`.
+		///
+		/// If the origin is Signed, then funds of signer are reserved according to the formula:
+		/// `MetadataDepositBase + MetadataDepositPerByte * (name.len + info.len)` taking into
+		/// account any already reserved funds.
+		///
+		/// - `class`: The identifier of the asset class whose instance's metadata to set.
+		/// - `instance`: The identifier of the asset instance whose metadata to set.
+		/// - `name`: The user visible name of this asset. Limited in length by `StringLimit`.
+		/// - `info`: The general information of this asset. Limited in length by `StringLimit`.
+		/// - `is_frozen`: Whether the metadata should be frozen against further changes.
+		///
+		/// Emits `MetadataSet`.
+		///
+		/// Weight: `O(1)`
+		#[pallet::weight(T::WeightInfo::set_metadata(name.len() as u32, info.len() as u32))]
+		pub(super) fn set_metadata(
+			origin: OriginFor<T>,
+			#[pallet::compact] class: T::ClassId,
+			#[pallet::compact] instance: T::InstanceId,
+			name: Vec<u8>,
+			info: Vec<u8>,
+			is_frozen: bool,
+		) -> DispatchResult {
+			let maybe_check_owner = T::ForceOrigin::try_origin(origin)
+				.map(|_| None)
+				.or_else(|origin| ensure_signed(origin).map(Some))?;
+
+			ensure!(name.len() <= T::StringLimit::get() as usize, Error::<T, I>::BadMetadata);
+			ensure!(info.len() <= T::StringLimit::get() as usize, Error::<T, I>::BadMetadata);
```

Comment:
would be nice if we could just already use bounded vec

Reply:
Another PR.